check/requiretls/requiretls.go

```diff
@@ -0,0 +1,26 @@
+package requiretls
+
+import (
+	"github.com/foxcpp/maddy/check"
+	"github.com/foxcpp/maddy/exterrors"
+	"github.com/foxcpp/maddy/module"
+)
+
+func requireTLS(ctx check.StatelessCheckContext) module.CheckResult {
+	if ctx.MsgMeta.SrcTLSState.HandshakeComplete {
+		return module.CheckResult{}
+	}
+
+	return module.CheckResult{
+		Reason: &exterrors.SMTPError{
+			Code:         550,
+			EnhancedCode: exterrors.EnhancedCode{5, 7, 1},
```

Comment:
More specific code for it is 5.7.10 and 523.

(from https://www.iana.org/assignments/smtp-enhanced-status-codes/smtp-enhanced-status-codes.xhtml)

Reply:
@foxcpp I did see that status code, but I think it's inappropriate to use here. The explanatory text is:

> This indicates that external strong privacy layer is needed in order to use the requested authentication mechanism. This is primarily intended for use with clear text authentication mechanisms. A client which receives this may activate a security layer such as TLS prior to authenticating, or attempt to use a stronger mechanism.

I read that to mean that it's only expected as a response to the `AUTH` command.

I picked 550 (and guessed 5.7.1), and the error text, based on this thread about what exchange does: https://social.technet.microsoft.com/Forums/ie/en-US/3d6cb80a-687e-407d-8cb4-e5c313f92757/email-bouncing-back-8220denied-by-policy-tls-conversation-required-diagnostic-code-smtp-550?forum=exchangesvrsecuremessaging

Happy to change it if my guess about 5.7.10 is wrong, though!

Reply:
Probably you are right. It is not a big deal, anyway.